### PR TITLE
[HZ-413] Introduce migration-tolerant IMap scan for Jet SQL engine

### PR DESCRIPTION
**Pull request by Fly-Style**
Checklist:
- [x] Labels (`Team:`, `Type:`, `Source:`, `Module:`) and Milestone set
- [x] Add `Add to Release Notes` label if changes should be mentioned in release notes or `Not Release Notes content` if changes are not relevant for release notes
- [x] Request reviewers if possible

### CODE REVIEW

File: hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/SqlProcessors.java [hunk 1]
```diff
@@ -46,4 +50,16 @@ public static ProcessorSupplier rowProjector(
                         SimpleExpressionEvalContext.from(ctx)));
         return mapUsingServiceP(service, RowProjector::project);
     }
+
+    public static ProcessorSupplier rowProjector(
```
Comment (gierlachg) [inline hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/SqlProcessors.java:54]: Either:
- move `RowProjectorProcessorSupplier` to _this_ package, if it's generic enough
- or remove _this_ method as it references `IMap` specific supplier or nothing is/will be using it

**Review by gierlachg — CHANGES_REQUESTED** [review]
If `OnHeapMapScanP` is not used anymore, why not to remove it together with its test, utils etc. ?

**Comment by Fly-Style** [comment]
> If `OnHeapMapScanP` is not used anymore, why not to remove it together with its test, utils etc. ?

It will be used in future. @viliam-durina could you provide your vision of its future usage?

**Review by viliam-durina — APPROVED** [review]

File: hazelcast/src/main/java/com/hazelcast/sql/impl/SqlDataSerializerHook.java [hunk 2]
```diff
@@ -200,11 +199,10 @@
 
     public static final int EXPRESSION_EXTRACT = 72;
 
-    public static final int MAP_SCAN_METADATA = 73;
 //    Reserved
-//    public static final int MAP_INDEX_SCAN_METADATA = 74;
+//    public static final int MAP_INDEX_SCAN_METADATA = 73;
```
Comment (gierlachg) [inline hazelcast/src/main/java/com/hazelcast/sql/impl/SqlDataSerializerHook.java:203]:
```suggestion
```

**Review by gierlachg — APPROVED** [review]

**Comment by Fly-Style** [comment]
run-lab-run

**Pull request merged**